hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HeapMemStoreLAB.java

```diff
@@ -107,6 +107,13 @@ public HeapMemStoreLAB(Configuration conf) {
   }
 
   /**
+   * To be used for merging multiple MSLABs
+   */
+  public void addPooledChunkQueue(BlockingQueue<PooledChunk> targetQueue) {
+    targetQueue.drainTo(pooledChunkQueue);
```

Comment:
What is the complexity here O(#items in queue)? How many items in queue, roughly? Could this become a bottleneck in merging/compaction?
Not important for this patch but maybe add a todo comment to change this to a set or list instead of queue